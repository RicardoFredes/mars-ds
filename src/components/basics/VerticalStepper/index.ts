/**
 * WARNING!
 * Do not change this file,
 * because it is automatically generated
 */

import VerticalStepper from "./vertical-stepper.component";
export default VerticalStepper;
export * from "./vertical-stepper.types";
