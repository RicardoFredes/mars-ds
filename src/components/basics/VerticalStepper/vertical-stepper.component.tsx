import type { VerticalStepperProps } from "./vertical-stepper.types";

import classNames from "classnames";

import Icon from "@/components/basics/Icon";
import Link from "@/components/basics/Link";
import Subtitle from "@/components/typographies/Subtitle";
import Text from "@/components/typographies/Text";

const VerticalStepper = ({ items, className, ...props }: VerticalStepperProps) => {
  const cn = classNames("vertical-stepper", className);

  return (
    <ol className={cn} {...props}>
      {items.map(
        (
          {
            title,
            subtitle,
            link,
            icon,
            iconColor = "var(--color-background-primary)",
            circleColor = "var(--color-text-primary)",
            children,
          },
          index
        ) => {
          const itemCn = classNames("vertical-stepper__item", {
            "vertical-stepper__item--has-icon": icon,
          });

          const containerCn = classNames("vertical-stepper__container", {
            "vertical-stepper__container--has-icon": icon,
          });

          const circleCn = classNames("vertical-stepper__circle", {
            "vertical-stepper__circle--has-icon": icon,
          });

          return (
            <li className={itemCn} key={index}>
              <Link {...link}>
                <div className={containerCn}>
                  <div
                    className={circleCn}
                    data-testid="vertical-stepper__circle"
                    style={{ backgroundColor: circleColor }}
                  >
                    {icon && (
                      <Icon
                        name={icon}
                        className="vertical-stepper__icon"
                        data-testid="vertical-stepper__icon"
                        color={iconColor}
                      />
                    )}
                  </div>

                  <div className="vertical-stepper__content">
                    <Text className="vertical-stepper__title" data-testid="vertical-stepper__title">
                      {title}
                    </Text>
                    {subtitle && (
                      <Subtitle
                        className="vertical-stepper__subtitle"
                        data-testid="vertical-stepper__subtitle"
                      >
                        {subtitle}
                      </Subtitle>
                    )}
                  </div>
                </div>
              </Link>

              {children && (
                <div
                  className="vertical-stepper__children"
                  data-testid="vertical-stepper__children"
                >
                  {children}
                </div>
              )}
            </li>
          );
        }
      )}
    </ol>
  );
};

export default VerticalStepper;
