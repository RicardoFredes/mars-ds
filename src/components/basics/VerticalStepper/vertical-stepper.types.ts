import type { HTMLProps, ReactNode } from "react";

import type { LinkProps } from "@/components/basics/Link";

interface VerticalStepperItemProps extends HTMLProps<HTMLLIElement> {
  title: string;
  link: LinkProps;
  subtitle?: string;
  children?: ReactNode;
  icon?: string;
  iconColor?: string;
  circleColor?: string;
}

interface VerticalStepperProps extends Omit<HTMLProps<HTMLOListElement>, "type"> {
  items: VerticalStepperItemProps[];
}

export type { VerticalStepperProps, VerticalStepperItemProps };
