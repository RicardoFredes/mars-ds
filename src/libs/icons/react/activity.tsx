/**
 * WARNING!
 * Do NOT edit this file as it is automatically generated
 * run "yarn generate:icons" to update
 */

  
import type { IconLibProps } from "@/libs/icons/icons.types";

const IconActivity = ({ height = 24, width = 24 }: IconLibProps) => (
<svg
  xmlns="http://www.w3.org/2000/svg"
  width={width}
  height={height}
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  strokeWidth="2"
  strokeLinecap="round"
  strokeLinejoin="round"
>
  <polyline points="22 12 18 12 15 21 9 3 6 12 2 12" />
</svg>
);
  
export default IconActivity;