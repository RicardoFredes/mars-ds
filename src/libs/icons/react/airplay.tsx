/**
 * WARNING!
 * Do NOT edit this file as it is automatically generated
 * run "yarn generate:icons" to update
 */

  
import type { IconLibProps } from "@/libs/icons/icons.types";

const IconAirplay = ({ height = 24, width = 24 }: IconLibProps) => (
<svg
  xmlns="http://www.w3.org/2000/svg"
  width={width}
  height={height}
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  strokeWidth="2"
  strokeLinecap="round"
  stroke-linejoin="round"
>
  <path d="M5 17H4a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h16a2 2 0 0 1 2 2v10a2 2 0 0 1-2 2h-1" />
  <polygon points="12 15 17 21 7 21 12 15" />
</svg>
);
  
export default IconAirplay;