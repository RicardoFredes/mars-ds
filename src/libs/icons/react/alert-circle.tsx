/**
 * WARNING!
 * Do NOT edit this file as it is automatically generated
 * run "yarn generate:icons" to update
 */

  
import type { IconLibProps } from "@/libs/icons/icons.types";

const IconAlertCircle = ({ height = 24, width = 24 }: IconLibProps) => (
<svg
  xmlns="http://www.w3.org/2000/svg"
  width={width}
  height={height}
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  strokeWidth="2"
  strokeLinecap="round"
  strokeLinejoin="round"
>
  <circle cx="12" cy="12" r="10" />
  <line x1="12" y1="8" x2="12" y2="12" />
  <line x1="12" y1="16" x2="12.01" y2="16" />
</svg>
);
  
export default IconAlertCircle;