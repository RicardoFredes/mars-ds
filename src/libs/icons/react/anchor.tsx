/**
 * WARNING!
 * Do NOT edit this file as it is automatically generated
 * run "yarn generate:icons" to update
 */

  
import type { IconLibProps } from "@/libs/icons/icons.types";

const IconAnchor = ({ height = 24, width = 24 }: IconLibProps) => (
<svg
  xmlns="http://www.w3.org/2000/svg"
  width={width}
  height={height}
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  strokeWidth="2"
  strokeLinecap="round"
  strokeLinejoin="round"
>
  <circle cx="12" cy="5" r="3" />
  <line x1="12" y1="22" x2="12" y2="8" />
  <path d="M5 12H2a10 10 0 0 0 20 0h-3" />
</svg>
);
  
export default IconAnchor;