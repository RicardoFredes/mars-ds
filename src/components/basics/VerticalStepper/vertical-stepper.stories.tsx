import type { ComponentMeta, ComponentStory } from "@storybook/react";

import VerticalStepper from "./vertical-stepper.component";

export default {
  title: "Components/Stepper/VerticalStepper",
  component: VerticalStepper,
  argTypes: {},
} as ComponentMeta<typeof VerticalStepper>;

const Template: ComponentStory<typeof VerticalStepper> = (args) => <VerticalStepper {...args} />;

const DefaultArgs = {
  items: [
    {
      title: "A soma de forma intuitiva",
      subtitle: "8 minutos",
      link: {
        href: "https://mesalva.com",
        target: "_blank",
      },
      icon: "video",
      iconColor: "var(--color-white)",
      circleColor: "var(--color-text-primary)",
    },
    {
      title: "Exercícios de compreensão sobre soma",
      subtitle: "3 exercícios",
      children: (
        <VerticalStepper
          items={[
            {
              title: "Exercício 1",
              link: {
                href: "https://mesalva.com",
                target: "_blank",
              },
              circleColor: "var(--color-text-disabled)",
            },
            {
              title: "Exercício 2",
              link: {
                href: "https://mesalva.com",
                target: "_blank",
              },
              circleColor: "var(--color-text-disabled)",
            },
            {
              title: "Exercício 3",
              link: {
                href: "https://mesalva.com",
                target: "_blank",
              },
              circleColor: "var(--color-text-disabled)",
            },
          ]}
        />
      ),
      icon: "edit",
      iconColor: "var(--color-white)",
      circleColor: "var(--color-text-primary)",
    },
    {
      title: "Somando de tudo",
      subtitle: "10 minutos",
      link: {
        href: "https://mesalva.com",
        target: "_blank",
      },
      icon: "video",
      iconColor: "var(--color-white)",
      circleColor: "var(--color-text-primary)",
    },
    {
      title: "Exercícios de compreensão sobre soma II",
      subtitle: "3 exercícios",
      link: {
        href: "https://mesalva.com",
        target: "_blank",
      },
      icon: "edit",
      iconColor: "var(--color-white)",
      circleColor: "var(--color-text-primary)",
    },
    {
      title: "Somando de tudo",
      subtitle: "9 minutos",
      link: {
        href: "https://mesalva.com",
        target: "_blank",
      },
      icon: "video",
      iconColor: "var(--color-white)",
      circleColor: "var(--color-text-primary)",
    },
    {
      title: "Exercícios de compreensão sobre subtração",
      subtitle: "3 exercícios",
      link: {
        href: "https://mesalva.com",
        target: "_blank",
      },
      icon: "edit",
      iconColor: "var(--color-white)",
      circleColor: "var(--color-text-primary)",
    },
    {
      title: "Exercícios de compreensão sobre subtração de naturais",
      subtitle: "5 exercícios",
      link: {
        href: "https://mesalva.com",
        target: "_blank",
      },
      icon: "edit",
      iconColor: "var(--color-white)",
      circleColor: "var(--color-text-primary)",
    },
  ],
};

export const Default = Template.bind({});
Default.args = DefaultArgs;
