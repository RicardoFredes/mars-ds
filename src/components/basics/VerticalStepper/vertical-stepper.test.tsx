import type { VerticalStepperProps, VerticalStepperItemProps } from "./vertical-stepper.types";

import { render } from "@testing-library/react";

import VerticalStepper from "./vertical-stepper.component";

const CLASSES = {
  item: "vertical-stepper__item",
  container: "vertical-stepper__container",
  circle: "vertical-stepper__circle",
  icon: "vertical-stepper__icon",
  content: "vertical-stepper__content",
  title: "vertical-stepper__title",
  subtitle: "vertical-stepper__subtitle",
  children: "vertical-stepper__children",
};

const defaultProps: VerticalStepperProps = {
  items: [
    {
      title: "A soma de forma intuitiva",
      link: {
        href: "https://mesalva.com",
        target: "_blank",
      },
    },
    {
      title: "Somando de tudo",
      subtitle: "10 minutos",
      link: {
        href: "https://mesalva.com",
        target: "_blank",
      },
      icon: "video",
      iconColor: "var(--color-white)",
      circleColor: "var(--color-text-primary)",
    },
  ],
};

const makeItemProps = (
  overrides: Partial<VerticalStepperItemProps> = {}
): VerticalStepperItemProps => ({
  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
  ...defaultProps.items[0]!,
  ...overrides,
});

const makeSut = (props?: VerticalStepperProps) =>
  render(<VerticalStepper {...defaultProps} {...props} />);

describe("<VerticalStepper>", () => {
  it("should match the snapshot", () => {
    const wrapper = makeSut();
    expect(wrapper).toMatchSnapshot();
  });

  it("should contain a link wrapping each item", () => {
    const wrapper = makeSut();
    const links = wrapper.getAllByRole("link") as HTMLAnchorElement[];

    expect(links).toHaveLength(defaultProps.items.length);
    expect(
      links.forEach((link) => {
        expect(link).toHaveAttribute("href");
        expect(link.attributes.getNamedItem("href")?.value).toBe(
          defaultProps.items?.[0]?.link.href
        );

        expect(link).toHaveAttribute("target");
        expect(link.attributes.getNamedItem("target")?.value).toBe(
          defaultProps.items?.[0]?.link.target
        );
      })
    );
  });

  describe("dots", () => {
    it("should contain a dot for each item", () => {
      const wrapper = makeSut();
      const dots = wrapper.getAllByTestId(CLASSES.circle);

      expect(dots.length).toBe(defaultProps.items.length);
    });

    it("should change color based on the item's circleColor prop", () => {
      const props: VerticalStepperProps = {
        items: [makeItemProps({ circleColor: "var(--color-text-disabled)" })],
      };

      const wrapper = makeSut(props);
      const dot = wrapper.getByTestId(CLASSES.circle);

      expect(dot).toHaveStyle(`background-color: ${props.items?.[0]?.circleColor}`);
    });
  });

  describe("when the icon prop is present", () => {
    it("should contain an icon", () => {
      const props: VerticalStepperProps = {
        items: [makeItemProps({ icon: "video" })],
      };
      const wrapper = makeSut(props);
      const icons = wrapper.getAllByTestId(CLASSES.icon);

      expect(icons.length).toBe(1);
    });

    it("should change color based on the item's iconColor prop", () => {
      const props: VerticalStepperProps = {
        items: [makeItemProps({ icon: "video", iconColor: "var(--color-text-disabled)" })],
      };

      const wrapper = makeSut(props);
      const icon = wrapper.getByTestId(CLASSES.icon);

      expect(icon).toHaveStyle(`background-color: ${props.items?.[0]?.circleColor}`);
    });
  });

  it("should contain a title", () => {
    const wrapper = makeSut();
    const titles = wrapper.getAllByTestId(CLASSES.title);

    expect(titles.length).toBe(defaultProps.items.length);
  });

  it("should contain a subtitle", () => {
    const props: VerticalStepperProps = {
      items: [makeItemProps({ subtitle: "8 minutos" })],
    };
    const wrapper = makeSut(props);
    const subtitles = wrapper.getAllByTestId(CLASSES.subtitle);

    expect(subtitles.length).toBe(1);
  });

  it("should contain children", () => {
    const props: VerticalStepperProps = {
      items: [makeItemProps({ children: <p>children</p> })],
    };
    const wrapper = makeSut(props);
    const children = wrapper.getAllByTestId(CLASSES.children);

    expect(children.length).toBe(1);
  });
});
