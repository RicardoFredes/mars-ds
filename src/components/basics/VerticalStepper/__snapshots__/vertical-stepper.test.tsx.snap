// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VerticalStepper> should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ol
        class="vertical-stepper"
      >
        <li
          class="vertical-stepper__item"
        >
          <a
            class="link"
            href="https://mesalva.com"
            target="_blank"
          >
            <div
              class="vertical-stepper__container"
            >
              <div
                class="vertical-stepper__circle"
                data-testid="vertical-stepper__circle"
              />
              <div
                class="vertical-stepper__content"
              >
                <p
                  class="text text--size-md vertical-stepper__title"
                  data-testid="vertical-stepper__title"
                >
                  A soma de forma intuitiva
                </p>
              </div>
            </div>
          </a>
        </li>
        <li
          class="vertical-stepper__item vertical-stepper__item--has-icon"
        >
          <a
            class="link"
            href="https://mesalva.com"
            target="_blank"
          >
            <div
              class="vertical-stepper__container vertical-stepper__container--has-icon"
            >
              <div
                class="vertical-stepper__circle vertical-stepper__circle--has-icon"
                data-testid="vertical-stepper__circle"
              >
                <span
                  class="icon vertical-stepper__icon"
                  data-testid="vertical-stepper__icon"
                >
                  <svg
                    fill="none"
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M21 7.15C20.3789 6.87035 19.6509 6.98841 19.15 7.45L17 9.45V8C17 6.34315 15.6569 5 14 5H5C3.34315 5 2 6.34315 2 8V16C2 17.6569 3.34315 19 5 19H14C15.6569 19 17 17.6569 17 16V14.55L19.16 16.55C19.478 16.8378 19.8911 16.9981 20.32 17C20.558 16.9994 20.7932 16.9483 21.01 16.85C21.6133 16.606 22.0087 16.0208 22.01 15.37V8.63C22.007 7.97674 21.6072 7.39089 21 7.15Z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </div>
              <div
                class="vertical-stepper__content"
              >
                <p
                  class="text text--size-md vertical-stepper__title"
                  data-testid="vertical-stepper__title"
                >
                  Somando de tudo
                </p>
                <h3
                  class="subtitle subtitle--size-md vertical-stepper__subtitle"
                  data-testid="vertical-stepper__subtitle"
                >
                  10 minutos
                </h3>
              </div>
            </div>
          </a>
        </li>
      </ol>
    </div>
  </body>,
  "container": <div>
    <ol
      class="vertical-stepper"
    >
      <li
        class="vertical-stepper__item"
      >
        <a
          class="link"
          href="https://mesalva.com"
          target="_blank"
        >
          <div
            class="vertical-stepper__container"
          >
            <div
              class="vertical-stepper__circle"
              data-testid="vertical-stepper__circle"
            />
            <div
              class="vertical-stepper__content"
            >
              <p
                class="text text--size-md vertical-stepper__title"
                data-testid="vertical-stepper__title"
              >
                A soma de forma intuitiva
              </p>
            </div>
          </div>
        </a>
      </li>
      <li
        class="vertical-stepper__item vertical-stepper__item--has-icon"
      >
        <a
          class="link"
          href="https://mesalva.com"
          target="_blank"
        >
          <div
            class="vertical-stepper__container vertical-stepper__container--has-icon"
          >
            <div
              class="vertical-stepper__circle vertical-stepper__circle--has-icon"
              data-testid="vertical-stepper__circle"
            >
              <span
                class="icon vertical-stepper__icon"
                data-testid="vertical-stepper__icon"
              >
                <svg
                  fill="none"
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M21 7.15C20.3789 6.87035 19.6509 6.98841 19.15 7.45L17 9.45V8C17 6.34315 15.6569 5 14 5H5C3.34315 5 2 6.34315 2 8V16C2 17.6569 3.34315 19 5 19H14C15.6569 19 17 17.6569 17 16V14.55L19.16 16.55C19.478 16.8378 19.8911 16.9981 20.32 17C20.558 16.9994 20.7932 16.9483 21.01 16.85C21.6133 16.606 22.0087 16.0208 22.01 15.37V8.63C22.007 7.97674 21.6072 7.39089 21 7.15Z"
                    fill="currentColor"
                  />
                </svg>
              </span>
            </div>
            <div
              class="vertical-stepper__content"
            >
              <p
                class="text text--size-md vertical-stepper__title"
                data-testid="vertical-stepper__title"
              >
                Somando de tudo
              </p>
              <h3
                class="subtitle subtitle--size-md vertical-stepper__subtitle"
                data-testid="vertical-stepper__subtitle"
              >
                10 minutos
              </h3>
            </div>
          </div>
        </a>
      </li>
    </ol>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
